app/controllers/locations_controller.rb

```diff
@@ -14,8 +15,144 @@ def index
     @searched = search_term.presence
   end
 
-  def new
-    @location = Location.new
+  def create
+    @location = @scheme.locations.create!
+    redirect_to scheme_location_postcode_path(@scheme, @location, route: params[:route])
+  end
+
+  def postcode; end
+
+  def update_postcode
+    @location.postcode = PostcodeService.clean(params[:location][:postcode])
+    if @location.postcode_changed?
+      @location.location_admin_district = nil
+      @location.location_code = nil
+    end
```

Comment:
```suggestion
    @location.postcode = PostcodeService.clean(params[:location][:postcode])
    if @location.postcode_changed?
      @location.location_admin_district = nil
      @location.location_code = nil
    end
```

This all belongs in the model.